### PR TITLE
[TI_RecordedFuture] Add a field to easily filter transform's source and destination indices

### PR DESCRIPTION
**Pull request by kcreddy**
## What does this PR do?
**Bakground:**
Adding indicator rules is a bit complex after IOC expiration support is launched. The destination indices (unexpired indicators) are needed to be queried using `_index` field and also using explicitly added Recorded Future and Anomali rules separately.

**This PR:**
- Adds a field onto source ingest pipeline as a placeholder separating source and destination indices.  
- This field is deleted inside destination ingest pipeline run by the transform. 

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

- [ ]

## Related issues

- Relates https://github.com/elastic/integrations/issues/6342

### CODE REVIEW

File: packages/ti_recordedfuture/data_stream/threat/fields/fields.yml [hunk 1]
```diff
@@ -19,3 +19,6 @@
       description: >
         Details of risk rules observed.
 
+- name: labels.is_ioc_transform_source
+  type: keyword
```
Comment (efd6) [inline packages/ti_recordedfuture/data_stream/threat/fields/fields.yml:23]: Same query; can this be a boolean?

File: packages/ti_recordedfuture/elasticsearch/transform/latest_ioc/transform.yml [hunk 2]
```diff
@@ -10,6 +10,9 @@ source:
 # time field type conflicts.
 dest:
   index: "logs-ti_recordedfuture_latest.threat-1"
+  # The version prefix on dest.pipeline corresponds to the package version.
+  # The version needs to be updated when package version changes.
```
Comment (efd6) [inline packages/ti_recordedfuture/elasticsearch/transform/latest_ioc/transform.yml:14]: Will this be caught be tests?

Comment (kcreddy) [reply packages/ti_recordedfuture/elasticsearch/transform/latest_ioc/transform.yml:14]: Yes, its caught in system tests

File: packages/ti_recordedfuture/data_stream/threat/fields/fields.yml (hunk 1 above)
Comment (kcreddy) [reply packages/ti_recordedfuture/data_stream/threat/fields/fields.yml:23]: Same reason as mentioned here: https://github.com/elastic/integrations/pull/6344#discussion_r1208933818

**Review by efd6 — APPROVED** [review]

**Review by andrewkroh — APPROVED** [review]

**Pull request merged**